presto-spi/src/main/java/io/prestosql/spi/predicate/EquatableValueSet.java

```diff
@@ -283,6 +302,16 @@ public String toString(ConnectorSession session)
                 .collect(toLinkedSet());
     }
 
+    private static <T> boolean intersectOverlaps(Set<T> set1, Set<T> set2)
+    {
+        for (T element : set1) {
```

Comment:
Iterate over the set with fewest elements.

Reply:
Made this change here and in `intersect` method